terraform/projects/app-ecs-nodes/main.tf

```diff
@@ -0,0 +1,179 @@
+/**
+* ## Project: app-ecs-nodes
+*
+* Create ECS worker nodes
+*
+*/
+
+variable "additional_tags" {
+  type        = "map"
+  description = "Stack specific tags to apply"
+  default     = {}
+}
+
+variable "aws_region" {
+  type        = "string"
+  description = "AWS region"
+  default     = "eu-west-1"
+} 
+
+variable "ecs_image_id" {
+  type        = "string"
+  description = "AMI ID to use for the ECS nodes"
+  default     = "ami-2d386654" # Latest Amazon ECS optimised AMI
+}
+
+variable "ecs_instance_type" {
+  type        = "string"
+  description = "ECS Node instance type"
+  default     = "t2.medium"
+}
+
+variable "ecs_instance_root_size" {
+  type        = "string"
+  description = "ECS instance root volume size - in GB"
+  default     = "50"
+}
+
+variable "ecs_instance_ssh_keyname" {
+  type        = "string"
+  description = "SSH keyname for ECS instances"
+  default     = "ecs-monitoring-ssh-test"
+}
+
+variable "remote_state_bucket" {
+  type        = "string"
+  description = "S3 bucket we store our terraform state in"
+  default     = "prometheus-staging"
+}
+
+variable "stack_name" {
+  type        = "string"
+  description = "Unique name for this collection of resources"
+  default     = "staging"
+}
+
+# locals
+# --------------------------------------------------------------
+
+locals {
+  default_tags = {
+    Terraform = "true"
+    "Project" = "app-ecs-nodes"
+  }
+
+  cluster_name = "${var.stack_name}-ecs-monitoring"
+}
+
+# Resources
+# --------------------------------------------------------------
+
+## Providers
+
+terraform {
+  required_version = "= 0.11.7"
+
+  backend "s3" {
+    key = "app-ecs-nodes.tfstate"
+  }
+}
+
+provider "aws" {
+  version = "~> 1.14.1"
+  region  = "${var.aws_region}"
+}
+
+provider "template" {
+  version = "~> 1.0.0"
+}
+
+## Data sources
+
+data "terraform_remote_state" "infra_networking" {
+  backend = "s3"
+
+  config {
+    bucket = "${var.remote_state_bucket}"
+    key    = "infra-networking.tfstate"
+    region = "${var.aws_region}"
+  }
+}
+
+data "terraform_remote_state" "infra_security_groups" {
+  backend = "s3"
+
+  config {
+    bucket = "${var.remote_state_bucket}"
+    key    = "infra-security-groups.tfstate"
+    region = "${var.aws_region}"
+  }
+}
+
+## Resources
+
+resource "null_resource" "node_autoscaling_group_tags" {
```

Comment:
Could someone please clarify why we need to do this? We appear to be doing tags in a different way for `monitoring_external_alb.tags` so trying to understand if there is intentional difference in behaviour?